dhctl/pkg/kubernetes/actions/manifests/manifests.go

```diff
@@ -611,25 +611,48 @@ func SecretWithStaticClusterConfig(configData []byte) *apiv1.Secret {
 	return generateSecret("d8-static-cluster-configuration", "kube-system", data, nil)
 }
 
-func SecretNameForNodeTerraformState(nodeName string) string {
-	return "d8-node-terraform-state-" + nodeName
+const maxLenSecretName = 63
+
+func SecretNameForNodeTerraformState(nodeName string) (string, error) {
```

Comment:
BTW, we have never expected before that the manifest function could fail.
If it were my decision, I'd validate the node name instead of a secret length because this is not a problem with the secret itself; it is a problem with the input.